platform/android/app/src/main/kotlin/coop/polypoly/polypod/homescreen/HomeScreenViewModel.kt

```diff
@@ -105,25 +109,49 @@ data class Footer(
 
 class HomeScreenViewModel {
     fun getSectionModels(
+        context: Context,
         onFeatureSelected: (String) -> Unit
     ): List<SectionModel> {
         return FeatureStorage.categories.map { category ->
             SectionModel(
                 category.name,
-                SectionType.fromCategoryType(category.category),
+                SectionType.fromCategoryType(category.id),
                 category.features.map { feature ->
                     TileModel(
                         feature.name,
-                        feature.description,
-                        feature.thumbnail,
+                        feature.description ?: "",
+                        createTileThumbnailBitmap(context, feature.thumbnail),
                         Color(feature.thumbnailColor),
                         Color(feature.borderColor),
-                        Color(feature.tileTextColor)
+                        Color(feature.tileTextColor),
                     ) {
                         onFeatureSelected(feature.id)
                     }
                 }
             )
         }
     }
+
+    private fun createTileThumbnailBitmap(
+        context: Context,
+        path: String?
+    ): Bitmap? {
+        if (path == null) {
+            return null
+        }
+        if (path.endsWith(".pdf")) {
+            return PDFBitmap
+                .bitmapFromPDF(
+                    File(path).inputStream(),
+                    context.resources.displayMetrics.densityDpi
+                )
+        } else {
+            val options = BitmapFactory.Options()
+            // For now, we assume all thumbnails are xhdpi, i.e. 2x scale factor
+            options.inDensity = DisplayMetrics.DENSITY_XHIGH
+            File(path).inputStream().use {
```

Comment:
Again, maybe call it with a `File`? In both cases path is turned into a file (with no try/catch); for the actual path `File` methods can be used.

Reply:
The Feature comes in with the thumbnailPath as string, it was choosing that mapping from String to File to be done in this function. But true, File can be created only once.